File: components/generate/src/test/java/org/cloudfoundry/credhub/requests/BulkRegenerateRequestTest.java
package org.cloudfoundry.credhub.requests;

import java.util.Set;

import javax.validation.ConstraintViolation;

import org.cloudfoundry.credhub.ErrorMessages;
import org.cloudfoundry.credhub.helpers.JsonTestHelper;
import org.hamcrest.MatcherAssert;
import org.hamcrest.Matchers;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class BulkRegenerateRequestTest {
  @Test
  public void whenSignedByValueIsMissing__isInvalid() {
    final Set<ConstraintViolation<BulkRegenerateRequest>> violations = JsonTestHelper.deserializeAndValidate("{}",
      BulkRegenerateRequest.class);

    MatcherAssert.assertThat(violations, Matchers.contains(JsonTestHelper.hasViolationWithMessage(ErrorMessages.MISSING_SIGNED_BY)));
  }

}

File: components/generate/src/test/kotlin/org/cloudfoundry/credhub/requests/BulkRegenerateRequestTest.kt
package org.cloudfoundry.credhub.requests

import org.cloudfoundry.credhub.ErrorMessages
import org.cloudfoundry.credhub.helpers.JsonTestHelper
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test

class BulkRegenerateRequestTest {
    @Test
    fun whenSignedByValueIsMissing__isInvalid() {
        val violations =
            JsonTestHelper.deserializeAndValidate(
                "{}",
                BulkRegenerateRequest::class.java,
            )

        MatcherAssert.assertThat(violations, Matchers.contains(JsonTestHelper.hasViolationWithMessage(ErrorMessages.MISSING_SIGNED_BY)))
    }

    @Test
    fun whenDurationProvided_deserializesDurationAndPrependsSignedBy() {
        val request =
            JsonTestHelper.createObjectMapper().readValue(
                """
                {
                  "signed_by": "some-ca",
                  "duration": 730
                }
                """.trimIndent(),
                BulkRegenerateRequest::class.java,
            )

        MatcherAssert.assertThat(request.signedBy, Matchers.equalTo("/some-ca"))
        MatcherAssert.assertThat(request.duration, Matchers.equalTo(730))
    }
}
